code/modules/reagents/chemistry/reagents/alchemical_potions.dm

```diff
@@ -0,0 +1,224 @@
+///////////////////////////////////////////////////////////////////
+					//Noita Potions
+//////////////////////////////////////////////////////////////////
+
+
+/datum/reagent/magic
+	name = "magic essence"
+	description = "Basic component in all magical potions."
+	random_unrestricted = FALSE
+
+//	----	POLYMORPHINE	----
+
+/datum/reagent/magic/polymorphine
+	name = "polymorphine"
+	description = "Transforms your target into a harmless animal."
+	color = "#cccccc"
+	taste_description = "a rainbow of tastes"
+	overdose_threshold = 30
+	metabolization_rate = 3 * REAGENTS_METABOLISM
+	var/list_mobs = FRIENDLY_SPAWN
+	var/obj/shapeshift_holder/shapeshiftdata
+
+/datum/reagent/magic/polymorphine/reaction_mob(mob/living/M, method=TOUCH, reac_volume)
+	if(method == TOUCH || method == VAPOR)
+		polymorph_target(M)
+	..()
+
+/datum/reagent/magic/polymorphine/overdose_start(mob/living/L)
+	..()
+	L.reagents.remove_reagent(/datum/reagent/magic/polymorphine,overdose_threshold)
+	polymorph_target(L)
+
+/datum/reagent/magic/polymorphine/proc/polymorph_target(mob/living/L)
+	shapeshiftdata = locate() in L
+	if(shapeshiftdata)		//user already polymorphed
+		return
+	var/mob/living/shape = create_random_mob((get_turf(L)), list_mobs)
+	shapeshiftdata = new(shape,null,L)
+	addtimer(CALLBACK(src, .proc/restore_target, L), 15 SECONDS)
+
+/datum/reagent/magic/polymorphine/proc/restore_target(mob/living/L)
+	. = ..()
+	if(!shapeshiftdata)
+		return
+	shapeshiftdata.restore()
+
+/datum/reagent/magic/polymorphine/chaotic
+	name = "chaotic polymorphine"
+	description = "Transforms your target into a hostile creature."
+	list_mobs = HOSTILE_SPAWN
+
+/datum/reagent/magic/polymorphine/unstable
+	name = "unstable polymorphine"
+	description = "Nobody knows what will happen!"
+
+/datum/reagent/magic/polymorphine/unstable/on_mob_metabolize(mob/living/L)
+	..()
+	list_mobs = prob(50) ? HOSTILE_SPAWN : FRIENDLY_SPAWN
+
+//	----	BERSERKIUM	----
+
+/datum/reagent/magic/berserkium
+	name = "polymorphine"
+	description = "Transforms your target into a harmless animal."
+	color = "#cccccc"
+	taste_description = "something that makes you angry!"
+	metabolization_rate = 3 * REAGENTS_METABOLISM
+
+/datum/reagent/magic/berserkium/on_mob_metabolize(mob/living/L)
+	. = ..()
+	ADD_TRAIT(L, TRAIT_IGNOREDAMAGESLOWDOWN, REAGENT_TRAIT)
+	ADD_TRAIT(L, TRAIT_NOSTAMCRIT, REAGENT_TRAIT)
+	ADD_TRAIT(L, TRAIT_NOLIMBDISABLE, REAGENT_TRAIT)
```

Comment:
Reagent trait is too general, especially for something like these traits which are used in reagents a lot. Make one specifically for beserkium.